### DISCUSSION TITLE
Deployment

### DISCUSSION THREAD

**Discussion by ademmler** [libvips/libvips#2035] [Q&A]
Hi there.
What is the best way to deploy my application which uses vips? 
I want to build "standalone applications", which includes libvips.
I want to do this for Windows and Mac OS X. 

Is there any tutorial or example?

Regards Alex

**Comment by kmartinez** [comment]
if you want cross-compatibility and use built-in vips library functionality - 
try pyvips. You can use a python GUI if you need an interface.
If you're doing C then the start-up learning curve will be trickier on Windows (I avoid it).
There is sample code for pyvips and starting with C - on our wiki/docs areas.
good luck!

**Comment by jcupitt** [comment]
I package nip2 for win and macos. Windows is fairly easy: just put the DLLs into the bin area. This uses that nullsoft installer, I expect you'd want something better.

https://github.com/libvips/build-win32

macOS is harder: you need a wrapper script. Have a look inside the `.app`.

https://github.com/jcupitt/build-osx

**Comment by kmartinez** [comment]
and nip2 is much faster for rapid prototyping of your application!

**Comment by ademmler** [comment]
Sorry - I have forgotten to mention that I use Qt and C++ for coding.  Payton is nice - but not on the list right now.
My question was more about should I compile static libs, what about all the dependencies, are there rebuild ready libvips ....
And finally about deployment to unknown hosts (non development) ... hence looking for your experiences. tips and tricks  ...

Regards Alexander

**Comment by ademmler** [comment]
> and nip2 is much faster for rapid prototyping of your application!

Nice hint - but I am beyond prototyping.
